doc/sphinx/language/core/assumptions.rst

```diff
@@ -38,6 +38,13 @@ variable can be introduced at the same time. It is also possible to give
 the type of the variable as follows:
 :n:`(@ident : @type := @term)`.
 
+`(x : T | P)` is syntactic sugar for `(x : Stdlib.Init.Specif.sig (fun x : T => P))`,
+which would more typically be written `(x : {x : T | P})`.
+Since `(x : T | P)` uses `sig` directly,
+changing the notation `{x : T | P}`
+will not change the meaning of `(x : T | P)`, while
+changing the implicit arguments of `sig` will break `(x : T | P)`).
+
```

Comment:
What about making it syntactic sugar for `@Stdlib.Init.Specif.sig T (fun x : T => P)` so that it becomes insensitive to the change of implicit arguments?

Reply:
Not in this PR

Reply:
Ok. Will you do one? Would you like I do?

Reply:
you can do it